merkledag.go

```diff
@@ -28,6 +27,8 @@ func (e ErrNotFound) Error() string {
 
 // Is allows to check whether any error is of this ErrNotFound type.
 // Do not use this directly, but rather errors.Is(yourError, ErrNotFound).
+// For maximum compatibility you should prefer IsNotFound() instead as it will
+// also match other compatible NotFound error types.
```

Comment:
is there a reason not to attempt the same duck-typing here as in IsNotFound?

Reply:
I'm on the fence about this one, my thinking is that when you're doing an `errors.Is()` you have an expectation that the thing your checking _IS_ the thing you are looking for, or at a minimum it's in the chain of wrapped errors somewhere. Changing it to check for `NotFound()` means that you're loosening that contract to be _IS LIKE_. I'm more comfortable doing it with `IsNotFound()` because that contract is local, not at the standard library level and because it has a very particular meaning - "is not found!".

But I could go the other way too, I can't think up a hypothetical case where you would want to differentiate. Maybe if you were checking to see whether you have an old-style Blockstore somewhere in your call stack .. for some reason?

@masih care to weigh in too? I don't mind going either way on this.

Reply:
Thinking out loud:

If the rationale is backward compatibility then my vote would be to have less verbose code and let `Is` just work without strict type matching. 

As far as the go documentation on this interface goes, it reads very loose to me; the documentation does not explicitly dictate that the type should match. In fact it specifically uses loose language like ["An error type might provide an Is method so it can be treated as equivalent to an existing error.".](https://pkg.go.dev/errors#example-Is).

Reply:
"equivalent" is the key word here, good catch that it's in the official docs!

I'll switch it out here and in go-ipld-prime so we get maximal compatibility and happiness.